sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -62,6 +62,7 @@ object Literal {
     case b: Byte => Literal(b, ByteType)
     case s: Short => Literal(s, ShortType)
     case s: String => Literal(UTF8String.fromString(s), StringType)
+    case s: UTF8String => Literal(s, StringType)
```

Comment:
IIUC, `Literal.apply` is for external values. If you have internal values like `UTF8String`, you should use `Literal(value, dt)`.

Reply:
Yeah, let's don't do this here.

Reply:
OK, tried that all case can be handled. Close this one